feature-toggles/feature-toggles-impl/src/test/java/com/duckduckgo/feature/toggles/codegen/ContributesRemoteFeatureCodeGeneratorBucketAssignmentTest.kt

```diff
@@ -62,12 +62,19 @@ class ContributesRemoteFeatureCodeGeneratorBucketAssignmentTest(private val test
 
     @Test
     fun `test probability`() {
-        fun isWithinRange(): Boolean {
-            if (testCase.run < 9000) return true // small sample sizes will have high variance
+        fun assertWithinRange() {
+            if (testCase.run < 9000) return // small sample sizes will have high variance
             val allowableDeviation = (enables + disables) / 2 * 6.0f / 100 // allowable deviation of 6%
 
             // Check if the absolute difference between the two numbers is within the allowable deviation
-            return abs(enables - disables) <= allowableDeviation
+            val difference = abs(enables - disables)
+            assertTrue(
+                "Given sample size of ${enables + disables}, the allowable deviation (6%) is: $allowableDeviation, " +
+                    "but actual difference was $difference " +
+                    "(enables=$enables" +
+                    ", disables=$disables)",
+                difference <= allowableDeviation,
+            )
```

Comment:
just adding more logging to the failures in case that helps highlight the problem